.github/workflows/_binary-build-linux.yml

```
@@ -163,17 +170,34 @@ jobs:
           repository: pytorch/builder
           path: builder
           quiet-checkout: true
+
       - name: Clean pytorch/builder checkout
         run: |
           # Remove any artifacts from the previous checkouts
           git clean -fxd
         working-directory: builder
 
+      - name: Check if the job is disabled
+        if: ${{ github.event.workflow_run.head_branch != 'nightly' }}
+        id: filter
+        uses: ./pytorch/.github/actions/filter-test-configs
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # NB: Use a mock test matrix with a default value here. After filtering, if the
+          # returned matrix is empty, it means that the job is disabled
+          test-matrix: |
+            { include: [
+              { config: "default" },
+            ]}
```

instead of always checking for `github.event.workflow_run.head_branch == 'nightly'` everywhere, what if you modified the filter to include an extra "don't filter anything" input, and _that_ input could be checking for `github.event.workflow_run.head_branch == 'nightly'`

This would localize the exception cases to just one line, and every other part of the workflow could simply check for `steps.filter.outputs.is-test-matrix-empty == 'False'`

It could look something like:

```suggestion
        with:
          disable-filter: ${{ github.event.workflow_run.head_branch == 'nightly' }}
          github-token: ${{ secrets.GITHUB_TOKEN }}
          # NB: Use a mock test matrix with a default value here. After filtering, if the
          # returned matrix is empty, it means that the job is disabled
          test-matrix: |
            { include: [
              { config: "default" },
            ]}
```

---

Yeah, this could be changed.  I have a prototype in which the head branch is passed into the script.  The nightly check could be done there instead.